packages/server/src/messages/api.ts

```diff
@@ -1,178 +1,156 @@
 import { uuid } from '@botpress/messaging-base'
-import { Router } from 'express'
+import { Response } from 'express'
 import { v4 as uuidv4 } from 'uuid'
-import { Auth } from '../base/auth/auth'
+import { ApiManager } from '../base/api-manager'
+import { ClientApiRequest } from '../base/auth/client'
 import { ConversationService } from '../conversations/service'
 import { ConverseService } from '../converse/service'
 import { Schema } from './schema'
 import { MessageService } from './service'
 
 export class MessageApi {
   constructor(
-    private router: Router,
-    private auth: Auth,
     private conversations: ConversationService,
     private messages: MessageService,
     private converse: ConverseService
   ) {}
 
-  async setup() {
-    this.router.post(
-      '/messages',
-      this.auth.client.auth(async (req, res) => {
-        const { error } = Schema.Api.Create.validate(req.body)
-        if (error) {
-          return res.status(400).send(error.message)
+  setup(router: ApiManager) {
+    router.post('/messages', Schema.Api.Create, this.create.bind(this))
+    router.post('/messages/collect', Schema.Api.Collect, this.collect.bind(this))
+    router.get('/messages/:id', Schema.Api.Get, this.get.bind(this))
+    router.get('/messages/conversation/:conversationId', Schema.Api.List, this.list.bind(this))
+    router.delete('/messages/:id', Schema.Api.Delete, this.delete.bind(this))
+    router.delete(
+      '/messages/conversation/:conversationId',
+      Schema.Api.DeleteByConversation,
+      this.deleteByConversation.bind(this)
+    )
+    router.post('/messages/turn/:id', Schema.Api.Turn, this.turn.bind(this))
+  }
+
+  async create(req: ClientApiRequest, res: Response) {
+    const { conversationId, authorId, payload, incomingId } = req.body as {
+      conversationId: uuid
+      authorId: uuid
+      payload: any
+      incomingId: uuid
+    }
+
+    const conversation = await this.conversations.get(conversationId)
+    if (!conversation || conversation.clientId !== req.client.id) {
+      return res.sendStatus(404)
```

Comment:
We return 404 if you're trying to use a ressource that either doesn't exist or doesn't belong to you. We return undefined if you're fetching a single ressource. So the only case where we return undefined instead of 404 is in the get function because it's valid for the api to lookup if a ressource exist that way. On the other hand, trying to create a message for a conversation that doesn't exist can't return anything else than an error because it's simply wrong api usage.